### ISSUE TITLE
Rule proposal: `no-class`

### ISSUE THREAD

**Issue by silverwind** [sindresorhus/eslint-plugin-unicorn#2075]
### Description

It might sound controversial to some, but I'm of the opinion that my code should not contain any `class` declarations as those are better written as `function`. I would like to see a rule similar to [this implementation](https://github.com/jfmengels/eslint-plugin-fp/blob/master/rules/no-class.js).

See [this issue](https://news.ycombinator.com/item?id=35747480) related to classes where they break language features like `Proxy`.

### Fail

```js
class foo {}
```
```js
const foo = class {}
```

### Pass

```js
function foo() {}
```
```js
const foo = () => {}
```

### Additional Info

The rule should likely be disabled by default.

**Comment by silverwind** [comment]
Actually this is possible already using [no-restricted-syntax](https://eslint.org/docs/latest/rules/no-restricted-syntax) and does almost exactly what I need, so closing.
